spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultTopicResolver.java

```diff
@@ -34,15 +41,37 @@
  *
  * @author Chris Bono
  * @author Aleksei Arsenev
+ * @author Jonas Geiregat
  */
-public class DefaultTopicResolver implements TopicResolver {
+public class DefaultTopicResolver implements TopicResolver, ApplicationContextAware {
+
+	private final LogAccessor logger = new LogAccessor(this.getClass());
 
 	private final Map<Class<?>, String> customTopicMappings = new LinkedHashMap<>();
 
 	private final PulsarMessageAnnotationRegistry pulsarMessageAnnotationRegistry = new PulsarMessageAnnotationRegistry();
 
 	private boolean usePulsarMessageAnnotations = true;
 
+	@Nullable
+	private ExpressionResolver expressionResolver;
+
+	/**
+	 * Create a new instance with the given expression resolver, that is able to resolve
+	 * SpEL expressions.
+	 * @param expressionResolver the expression resolver to use
+	 */
+	public DefaultTopicResolver(ExpressionResolver expressionResolver) {
+		this.expressionResolver = expressionResolver;
+	}
+
+	/**
+	 * Create a new instance with the default expression resolver without the ability to
```

Comment:
The default expression resolver will have the ability to resolve expressions, it just will not be some custom one passed in by the user. Longer term, SB will auto-configure the DTR and we can drop the BFA implementation at that time if we want.